share/spice/public_holidays/public_holidays.js

```diff
@@ -0,0 +1,64 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_public_holidays = function(api_result) {
+
+        if (!api_result || api_result.error || api_result.errors || !api_result.holidays || api_result.holidays.length === 0) {
+            return Spice.failed('public_holidays');
+        }
+
+        var script = $('[src*="/js/spice/public_holidays/"]')[0];
+        var source = $(script).attr("src");       
+        var year = decodeURIComponent(source.match(/public_holidays\/[a-z]{2}\/([0-9]{4})\/[^\/]+/)[1]);
+        var country = decodeURIComponent(source.match(/public_holidays\/[a-z]{2}\/[0-9]{4}\/([^\/]+)/)[1]);        
+        
+        if (!year || !country) {
+            return Spice.failed('public_holidays');                
+        }
+
+        DDG.require('moment.js', function() {
+            var data = {
+                title: "Public Holidays " + year, 
+                subtitle: country,
+                holidays: []
+            };
+            
+            $.each(api_result.holidays, function(index, holiday) {
+                // Some holidays only apply to certain provinces
+                var provinces = [];
+                if (holiday.states) {                    
+                    for (var i=0; i<holiday.states.length; ++i) {
+                        provinces.push(holiday.states[i].name);
+                    }
+                }
+                
+                data.holidays.push({
+                    name:  holiday.name,
+                    date:  moment(holiday.date.iso).format('ddd Do MMM'),
+                    notes: provinces.join(", ")
+                });
+            });
+            
+            // holiday.urlid is of the form "<country>/<holiday>"
+            var countryUrlId = api_result.holidays[0].urlid.match(/(.+)\/.*/)[1];            
+            
+            Spice.add({
+                id: "public_holidays",            
+                name: "Answer",
+                data: data,
+
```

Comment:
@sekhavati let's add `signal: "high",` -- this will force it to open this IA's tab over others'